torch/distributed/_spmd/parallel_mode.py

```diff
@@ -60,6 +60,7 @@ class DataParallel(ParallelMode):
     def __init__(
         self,
         parallel_style: str = "replicate",
+        input_batch_dim: int = 0,
```

Comment:
do we need a test to cover the case where batch dim is not dim0?

Reply:
Yep good point, let me try to add a test case in a separate PR to avoid trigger CI again